autorest/preparer.go

```diff
@@ -80,6 +80,16 @@ func WithNothing() PrepareDecorator {
 	}
 }
 
+// Nest returns a PrepareDecorator that nests the prepare decorators together.
+func Nest(pds ...PrepareDecorator) PrepareDecorator {
```

Comment:
If this makes sense for PrepareDecorators, I think it should make sense for ResponseDecorators as well. For that reason, I think we should add them both in the same PR and make sure their names don't collide. How about:

`func AggregatePreparerDecorator(pds ...PrepareDecorator) PrepareDecorator`
and
`func AggregateResponseDecorator(pds ...ResponseDecorator) ResponseDecorator`